ludwig/features/numerical_feature.py

```diff
@@ -81,17 +84,46 @@ def add_feature_data(
         dataset[feature[PROC_COLUMN]] = dataset_df[feature[COLUMN]].astype(
             np.float32).values
         if preprocessing_parameters['normalization'] is not None:
-            if preprocessing_parameters['normalization'] == 'zscore':
-                mean = metadata[feature[NAME]]['mean']
-                std = metadata[feature[NAME]]['std']
-                dataset[feature[PROC_COLUMN]] = (dataset[
-                                                     feature[
-                                                         PROC_COLUMN]] - mean) / std
-            elif preprocessing_parameters['normalization'] == 'minmax':
-                min_ = metadata[feature[NAME]]['min']
-                max_ = metadata[feature[NAME]]['max']
-                values = dataset[feature[PROC_COLUMN]]
-                dataset[feature[PROC_COLUMN]] = (values - min_) / (max_ - min_)
+            normalization_type = preprocessing_parameters['normalization']
+            NumericTransformer = get_from_registry(
+                normalization_type,
+                numeric_transformation_registry
+            )
+
+            if normalization_type == 'zscore':
+                numeric_transformer = NumericTransformer(
+                    metadata[feature[NAME]]['mean'],
+                    metadata[feature[NAME]]['std']
+                )
+            elif normalization_type == 'minmax':
+                numeric_transformer = NumericTransformer(
+                    metadata[feature[NAME]]['min'],
+                    metadata[feature[NAME]]['max']
+                )
+            elif normalization_type == 'log1p':
+                numeric_transformer = NumericTransformer()
+            else:
+                raise ValueError(
+                    'Normalization "{}" not supported. Valid values are '
+                    '"minmax" or "zscore"'.format(normalization_type)
+                )
```

Comment:
Thinking about ways to remove this if.
If the `NumericTransformer.__init__` function has `**kwargs`, we can pass the entire `metadata[feature[NAME]]` without problems.
So we sould just need:
```
NumericTransformer = get_from_registry(
                 preprocessing_parameters['normalization'],
                 numeric_transformation_registry
             )(**metadata[feature[NAME]])
```
and entirely remove this if. What do you think?

Reply:
done